src/java.base/share/conf/security/java.security

```diff
@@ -1171,6 +1171,19 @@ jceks.key.serialFilter = java.base/java.lang.Enum;java.base/java.security.KeyRep
 # name, an exception will be thrown when the property is used.
 # If the property is not set or empty, a default value will be used.
 #
+# For compatibility, the system property "keystore.pkcs12.legacy" can be set
```

Comment:
Was wondering if we should add why you might want to set this property, ex: "For compatibility with JDK or PKCS12 implementations that do not support the stronger algorithms ..." 

Compatibility with prior JDK versions should be less of an issue over time as these stronger settings and algs have been backported to prior JDKs.

Reply:
OpenSSL's help page shows
```
 -legacy             Use legacy encryption: 3DES_CBC for keys, RC2_CBC for certs
```
Can we also say "To work with legacy PKCS #12 files"?

Reply:
But isn't it mostly an issue when creating new keystores and not reading existing ones? I would want to avoid users thinking that they had to set this in more cases than needed.

Reply:
How about this?
```
To work with legacy PKCS #12 tools that does not support the new algorithms,
the system property "keystore.pkcs12.legacy" can be set
which will override the properties defined here with weaker algorithms.
This system property is equivalent to
```

Reply:
I think the text above might still make some users concerned that they should always set this property.
Maybe we can be less specific, and just say: "If you encounter compatibility issues with software that doesn't support the stronger algorithms, the system property ..."

Reply:
Can we say both? All these properties are only used when creating the file (key-related ones when creating the key). If a compatibility issue already happens, users need to downgrade their keystore.

So, the full text will be something like
```
To work with legacy PKCS #12 tools that does not support the new algorithms,
the system property "keystore.pkcs12.legacy" can be set
which will override the properties defined here with old settings.
If you encounter compatibility issues with software that doesn't support the stronger algorithms, you can downgrade the keystore with

   keytool -J-Dkeystore.pkcs12.legacy -importkeystore -keystore ks ...
```
I'll double check if the command can indeed downgrade key algorithms as well.

Reply:
It's a little long, but I can see why it is useful, so I think it's good. I would avoid the word "new" as this won't be new in a few years time. Here is an edit where I removed words which I thought were not essential:

> Some PKCS #12 tools and libraries may not support algorithms based on PBES2 and AES. 
> To create a PKCS #12 keystore which they can load, set the system property
> "keystore.pkcs12.legacy" which overrides the values of the properties defined below with
> legacy algorithms. Setting this system property is equivalent to
> 
>   ....
> 
> Also, you can downgrade an existing PKCS #12 keystore created with stronger algorithms
> to legacy algorithms with
> 
>    keytool -J-Dkeystore.pkcs12.legacy -importkeystore -srckeystore ks -destkeystore ks
> 
> This system property should be used at your own risk. 

Don't think you really need the sentence below, as you have already given several examples:

> Please note there is
> no value defined for this system property, i.e. "-Dkeystore.pkcs12.legacy"
> has the same effect as "-Dkeystore.pkcs12.legacy=<any value>".

Reply:
The reason I added the last sentence is because this property has no value. Someone might think they can set it to false to disable it, but that is equivalent to set it to true.

Reply:
Ah I see. Maybe put in the previous sentence, ex: "When set, this system property (which can only be enabled and has no value) is equivalent to:"

Just a suggestion.

Reply:
Suggested accepted. I just pushed a new commit. I modified "PKCS #12" to "PKCS12" because that's the word we used throughout the file.